src/posts/lucky_0.18.cr

```diff
@@ -0,0 +1,156 @@
+    for help with migrating your app.
+
+    ## What's new?
+
+    We've made a ton of changes for this release, but we will highlight a few.
+    You can see a full list through our [CHANGELOG](https://github.com/luckyframework/lucky/blob/master/CHANGELOG.md#changes-in-018).
+
+    ### Built in JWT (JSON Web Token) authentication
+
+    When generating a new app with `lucky init` you will be asked if you want
+    Lucky to include authentication. If you choose "yes" Lucky will now include
+    code for signing up users, and creating a token that can be used to authenticate
+    users with your API.
+
+    Read more about it in the [API Authentication guide](#{Guides::Authentication::Api.path}).
+
+    ### Improved serializers
+
+    Before Lucky 0.18.0 we recommended creating two serializers for each resource.
+    One for rendering a single item, and another for rendering a collection. This led
+    to a lot of files and code doing the same thing. Now Lucky generates a `BaseSerializer`
+    that includes a `for_collection` method.
+
+    That means you only need one class:
+
+    ```crystal
+    class UserSerializer < BaseSerializer
+      def initialize(@user : User)
+      end
+
+      def render
+        {id: @user.id, name: @user.name}
+      end
+    end
+    ```
+
+    And it can render one User or a collection of Users:
+
+    ```crystal
+    # Serialize one user
+    UserSerializer.new(UserQuery.first)
+
+    # Serialize a collection of users
+    UserSerializer.for_collection(UserQuery.new)
+    ```
+
+    Learn more in the [JSON rendering guide](#{Guides::JsonAndApis::RenderingJson.path(anchor: Guides::JsonAndApis::RenderingJson::ANCHOR_SERIALIZERS)})
+
+    ### Added `between` to queries
+
+    Added a new `between` criteria method to making querying easier:
+
+    ```crystal
+    UserQuery.new.created_at.between(30.days.ago, Time.utc)
+    ```
+
+    ### Improved error handling
+
+    Previously error handling had lots of conditional logic that made it hard to
+    tell what was happening. Now many errors are handled automatically, and
+    it is easier to change how an error is rendered to a user.
+
+    For example if you wanted to customize how `MyCustomError` was rendered:
+
+    ```crystal
+    class Errors::Show < Lucky::ErrorAction
+      def render(error : MyCustomError)
+        error_json "This is a custom error message", status: 418
+      end
+    end
+    ```
+
+    More in the [error handling guide](#{Guides::HttpAndRouting::ErrorHandling.path})
+
+    ### Error reporting
+
+    We've also introduced a way to easily report errors. Lucky generates
+    an `Errors::Show` class with a new `report` method that is empty. You can
+    fill in the method to report it however you want.
+
+    For example, we could use the [Raven shard](https://github.com/sija/raven.cr)
+    to send an error report to [Sentry](https://sentry.io/):
+
+    ```crystal
+    # src/actions/errors/show.cr
+    def report(error : Exception)
```

Comment:
```suggestion
    private def report(error : Exception) : Nil
```

When we generate this, it's a private method that returns nil. I think that `Raven.capture` does return nil, so the example is probably fine.